src/routes/clients.js

```diff
@@ -80,24 +80,24 @@ router.post('/', (req, res) => {
     return ApiError(res, 400, errno, BAD_REQUEST);
   }
 
+  let client;
   db().then(models => {
-    let client;
-    models.sequelize.transaction(transaction => {
+    return models.sequelize.transaction(transaction => {
       return models.Clients.create(req.body, { transaction }).then(_client => {
         client = _client;
         if (!req.body.permissions) {
           return;
         }
         return client.addPermissions(req.body.permissions, { transaction });
```

Comment:
you could do:
```javascript
return client.addPermissions(req.body.permissions, { transaction })
  .then(() => client);
```

and then below:
```javascript
.then(client => {
  res.status(201).send(client);
});
```

So that you don't need to have the `client` temp variable. The object is just passed along with the promise.